tbtaf/publisher/TBTAFPublisher.py

```diff
@@ -24,23 +29,26 @@ def PublishTestPlan(self, tBTestSuiteInstance, filePath, formatFlag):
         based on the discovered metadata
         '''
 
-        if formatFlag.lower()  != "html":
-            print "Format " + formatFlag + " is not supported"
+        if formatFlag.lower() not in TBTAFPublisher.FORMATS_SUPPORTED:
+            print("Format " + formatFlag + " is not supported")
             raise NonSupportedFormatException("NonSupportedFormatException in PublishTestPlan")
         
-        if not filePath.lower().endswith(".html"):
-            print "The file path doesn't contain a valid html file"
+        fileExtension = os.path.splitext(filePath.lower())[1]
+        if fileExtension not in TBTAFPublisher.FILE_EXTENSIONS_SUPPORTED :
+            print("The file path doesn't contain a valid " + formatFlag.lower() + " file")
             raise NonSupportedFormatException("NonSupportedFormatException in PublishTestPlan")
 
         try:
-            htmlFile = open(filePath,'w+')
+            mode = 'w+' if formatFlag.lower() == "html" else 'w+b'
+            htmlFile = open(filePath, mode)
         except IOError:
-            print "Invalid filePath"
+            print("Invalid filePath")
```

Comment:
Reword as "Invalid file path"

Reply:
Fixed in commit 34e2ef8c44f160aa6a8191335db238ae83e27923